lib/Test/Nginx/Socket.pm

```diff
@@ -804,6 +804,30 @@ again:
 
     test_stap($block, $dry_run);
 
+    if (defined $block->stop_after_request) {
+        my $ngx_pid = get_pid_from_pidfile($name);
+        my $TestNginxSleep = sleep_time();
+
+        if (is_running($ngx_pid)) {
+            kill(SIGQUIT, $ngx_pid);
+            waitpid($ngx_pid, 0);
+
+            sleep $TestNginxSleep;
+        }
+
+        my $max_i = 15;
```

Comment:
I don't like the `max_i` variable name. `max_tries` is more descriptive, for example.